store/auth.go

```diff
@@ -166,3 +173,130 @@ func DischargeAuthCaveat(username, password, macaroon, otp string) (string, erro
 	}
 	return responseData.Macaroon, nil
 }
+
+func requestDeviceNonce() ([]byte, error) {
+	const errorPrefix = "cannot get device identity nonce from store: "
+
+	req, err := http.NewRequest("POST", DeviceIdentityAPI+"/nonces", nil)
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Set("Accept", "application/json")
+
+	client := &http.Client{}
```

Comment:
There's a global client that has timeouts properly set in this package, I believe.